### PR TITLE
fix issue #2077

### PR DESCRIPTION
**Pull request by langghaha**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix issue [#2077](https://github.com/alibaba/nacos/issues/2077)

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java [hunk 1]
```diff
@@ -114,6 +114,14 @@ public Boolean publishConfig(HttpServletRequest request, HttpServletResponse res
         ParamUtils.checkParam(dataId, group, "datumId", content);
         ParamUtils.checkParam(tag);
 
+        //配置回滚时未接收到 type ,以免导致回滚后 type 字段值丢失
```
Comment (nkorange) [inline config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java]: 这句注释有一点模糊

Comment (nkorange) [reply config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java]: 你是指控制台没有传入type参数吗？

Comment (langghaha) [reply config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java]: 是的 前台没有传type参数到后台 导致保存时缺失type字段

Comment (langghaha) [reply config/src/main/java/com/alibaba/nacos/config/server/controller/ConfigController.java]: image
因为`type`为`null`导致insert后值被丢失

**Pull request closed without merging**